DemoServer/Controllers/Demos/MultiMapIndexes/MultiMapReduceIndex/MultiMapReduceIndexController.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System.Threading.Tasks;
+using DemoCommon.Models;
+using DemoServer.Utils.Cache;
+using DemoServer.Utils.Database;
+using DemoServer.Utils.UserId;
+using Microsoft.AspNetCore.Mvc;
+#region Usings
+using System.Linq;
+using Raven.Client.Documents.Linq;
+using System.Collections.Generic;
+using Raven.Client.Documents.Session;
+using Raven.Client.Documents.Indexes;
+#endregion
+
+namespace DemoServer.Controllers.Demos.MultiMapIndexes.MultiMapReduceIndex
+{
+    public class MultiMapReduceIndexController : DemoCodeController
+    {
+        public MultiMapReduceIndexController(UserIdContainer userId, UserStoreCache userStoreCache, MediaStoreCache mediaStoreCache,
+            DatabaseSetup databaseSetup) : base(userId, userStoreCache, mediaStoreCache, databaseSetup)
+        {
+        }
+        
+        #region Demo
+        #region Step_1
+        public class CityCommerceDetails : AbstractMultiMapIndexCreationTask<CityCommerceDetails.IndexEntry>
+        #endregion
+        {
+            #region Step_2
+            public class IndexEntry
+            {
+                public string CityName { get; set; }
+                public int NumberOfCompaniesInCity { get; set; } 
+                public int NumberOfSuppliersInCity { get; set; }
+                public int NumberOfOrdersShippedToCity { get; set; }
+            }
+            #endregion
+
+            public CityCommerceDetails()
+            {
+                #region Step_3
+                AddMap<Company>(companies => from company in companies
+                    select new IndexEntry
+                    {
+                        CityName = company.Address.City,
+                        NumberOfCompaniesInCity = 1,
+                        NumberOfSuppliersInCity = 0,
+                        NumberOfOrdersShippedToCity = 0
+                    });
+                
+                AddMap<Supplier>(suppliers => from supplier in suppliers
+                    select new IndexEntry
+                    {
+                        CityName = supplier.Address.City,
+                        NumberOfCompaniesInCity = 0,
+                        NumberOfSuppliersInCity = 1,
+                        NumberOfOrdersShippedToCity = 0
+                    });
+                
+                AddMap<Order>(orders => from order in orders
+                    select new IndexEntry
+                    {
+                        CityName = order.ShipTo.City,
+                        NumberOfCompaniesInCity = 0,
+                        NumberOfSuppliersInCity = 0,
+                        NumberOfOrdersShippedToCity = order.Lines.Count
```

Comment:
The property name is "NumberOfOrdersShippedToCity" but what we count here is not a number of orders - but the number of order lines. Maybe we could count order lines quantities for every line, and then finally come up with a count of items shipped to a city?

Reply:
I see - will use `NumberOfItemsShippedToCity` instead

Reply:
done